app/app.go

```diff
@@ -1151,16 +1155,108 @@ func (app *App) InitChainer(ctx sdk.Context, req abci.RequestInitChain) abci.Res
 }
 
 func (app *App) PrepareProposalHandler(ctx sdk.Context, req *abci.RequestPrepareProposal) (*abci.ResponsePrepareProposal, error) {
-	if app.mevHandler != nil {
-		return app.mevHandler.Handle(ctx, req)
+	// Get all pending bundles for this height
+	ctx.Logger().Debug("Preparing proposal", "height", ctx.BlockHeight())
+
+	if !app.mevConfig.Enabled {
+		return &abci.ResponsePrepareProposal{
+			TxRecords: utils.Map(req.Txs, func(tx []byte) *abci.TxRecord {
+				return &abci.TxRecord{Action: abci.TxRecord_UNMODIFIED, Tx: tx}
+			}),
+		}, nil
+	}
+
+	bundles := app.MevKeeper.PendingBundles(ctx.BlockHeight())
```

Comment:
I'd have a function/method in x/mev that has all of the code that follows this point.

Reply:
Good idea! Unfortunately having this in a separate module would create cyclic dependency (App needs to call a method, and this method needs access to App structure). But I moved it to a separate function, it's def cleaner